go/vt/vtgate/engine/projection.go

```diff
@@ -88,9 +88,10 @@ func (p *Projection) TryStreamExecute(ctx context.Context, vcursor VCursor, bind
 	env := evalengine.NewExpressionEnv(ctx, bindVars, vcursor)
 	var once sync.Once
 	var fields []*querypb.Field
+	var mu sync.Mutex
 	return vcursor.StreamExecutePrimitive(ctx, p.Input, bindVars, wantfields, func(qr *sqltypes.Result) error {
 		var err error
-		if wantfields {
+		if wantfields && qr.Fields != nil {
```

Comment:
@harshit-gangal This change breaks a bunch of tests it looks like. Why was this changed?

Reply:
It looks like this change was made due to the testing in `projection_test.go`, but I suspect it's a problem with that test? Since it runs concurrent stream results, but even if that happens, it's guaranteed that there would always be a field result first, so I don't think the test models a real scenario then.

Reply:
I also think that now that we lock here to also protect the evalengine expression env, we don't need the more elaborate locking around `wantfields` anymore.